src/evaluators/interpolation/PHAL_NodesToCellInterpolation_Def.hpp

```diff
@@ -13,10 +13,25 @@ namespace PHAL {
 template<typename EvalT, typename Traits, typename ScalarT>
 NodesToCellInterpolationBase<EvalT, Traits, ScalarT>::
 NodesToCellInterpolationBase (const Teuchos::ParameterList& p,
-                              const Teuchos::RCP<Albany::Layouts>& dl) :
-  BF        (p.get<std::string>("BF Variable Name"), dl->node_qp_scalar),
-  w_measure (p.get<std::string>("Weighted Measure Name"), dl->qp_scalar)
+                              const Teuchos::RCP<Albany::Layouts>& dl)
 {
+  std::string interpType = p.get<std::string>("Interpolation Type");
+  TEUCHOS_TEST_FOR_EXCEPTION ((interpType != "Value At Cell Barycenter") && (interpType != "Cell Average"),
+      std::runtime_error, "Interpolation Type can either be \"Value At Cell Barycenter\" or \"Cell Average\"");
+
+  interpolationType = (interpType == "Value At Cell Barycenter") ?
+      ValueAtCellBarycenter : CellAverage;
+
+  if(interpolationType == ValueAtCellBarycenter) {
+    onePointCubature = p.get<Teuchos::RCP <Intrepid2::Cubature<PHX::Device> > >("One Point Cubature");
```

Comment:
See comment above. The user should not have to pass the cubature. If, during merges or other refactors, or for whatever reason, someone passes a cubature that is not a 1pt cubature, we would not catch the error. Sure, we can put a Teuchos test to check it, but why bother if we can build the cubature internally?

Reply:
agreed